crates/bifrost/src/providers/replicated_loglet/provider.rs

```diff
@@ -349,7 +361,18 @@ impl<T: TransportConnect> LogletProvider for ReplicatedLogletProvider<T> {
 
         match selection {
             Ok(nodeset) => {
-                debug_assert!(nodeset.len() >= defaults.replication_property.num_copies() as usize);
+                let mut node_set_checker =
+                    NodeSetChecker::new(&nodeset, &nodes_config, &defaults.replication_property);
+                node_set_checker.fill_with(true);
+
+                // check that the new node set fulfills the replication property
+                if !node_set_checker.check_write_quorum(|attr| *attr) {
+                    // we couldn't find a nodeset that fulfills the desired replication property
+                    return Err(OperationError::retryable(InsufficientNodesError(
```

Comment:
I'm not sure if this should be considered retryable. Retryable imply a transient error, which is not the case here.